build_defs/antlr.bzl

```diff
@@ -0,0 +1,54 @@
+#-----------------------------------------------------------------------------
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-----------------------------------------------------------------------------
+"""
+Generates C++ parser, visitor, listener, lexer from a grammar file.
+"""
+
+load("@rules_antlr//antlr:antlr4.bzl", "antlr")
+
+def antlr_cc_library(name, grammar_file, package = None, listener = True, visitor = True):
+    """Creates a C++ lexer, visitor, listener, parser from a source grammar.
+    Args:
+      name: Base name for the lexer and the parser rules.
+      src: source ANTLR grammar file
+      package: The namespace for the generated code
+      listener: generate ANTLR listener (default: True)
+      visitor: generate ANTLR visitor (default: True)
+    """
+    generated = name + "_grammar"
+    antlr(
+        name = generated,
+        srcs = [grammar_file],
+        language = "Cpp",
+        listener = listener,
+        visitor = visitor,
+        package = "",
+    )
```

Comment:
Are we actually using the listener? I thought that its interface wasn't a good fit for us and we are only going to use the visitor.

Reply:
No. Good catch. I am going to disable it.